src/find_boundaries.jl

```diff
@@ -0,0 +1,114 @@
+                         background = zero(T)
+                        ) where T
+    find_boundaries!(copy(img), region; background = background)
+end
+
+## Alternate implementations using dilate and erode
+# The implementations below are based on scikit-image
+
+#=
+Copyright (C) 2021, Mark Kittisopikul and JuliaImages Github Organization
+Copyright (C) 2019, the scikit-image team
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+ 1. Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+ 2. Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in
+    the documentation and/or other materials provided with the
+    distribution.
+ 3. Neither the name of skimage nor the names of its contributors may be
+    used to endorse or promote products derived from this software without
+    specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+=#
+
+function find_boundaries_thick_dilate_erode(img::AbstractArray, region = coords_spatial(img))
+    # https://github.com/scikit-image/scikit-image/blob/d44ceda6241cb23a22dc8abf09a05090ed14da7f/skimage/segmentation/boundaries.py#L165-L167
+    return dilate(img, region) .!= erode(img, region)
+end
+function find_boundaries_dilate_erode(img::AbstractArray{T},
+                                      region = coords_spatial(img);
+                                      background = zero(T)
+                                     ) where T
+    # https://github.com/scikit-image/scikit-image/blob/d44ceda6241cb23a22dc8abf09a05090ed14da7f/skimage/segmentation/boundaries.py#L165-L170
+    thick_boundaries = find_boundaries_thick_dilate_erode(img, region)
+    foreground_img = img .!= background
+    return thick_boundaries .& foreground_img
+end
+"""
+    find_boundaries_thick(img::AbstractArray, [region,])
+
+    Find thick boundaries that are just outside and just inside the objects.
+    `region` indicates which dimensions to detect boundaries along.
+"""
+const find_boundaries_thick = find_boundaries_thick_dilate_erode
```

Comment:
The equation above differs from scikit-image. I have the original text. The equation says that the boundary of A is the set difference of A and the erosion of A by a structuring element B. For a 3x3 structuring element of 1's that we are assuming, this should result in a 1-pixel wide boundary.

scikit-image defines a "thick boundary" which is distinct from the above. The thick boundary is actual the default mode used by [`find_boundaries` in scikit-image](https://github.com/scikit-image/scikit-image/blob/0f1816fbcac2e6be7c9ff4cf549c45e49aef2ce1/skimage/segmentation/boundaries.py#L49-L188). I have included it here for compatibility with scikit-image and since other pull requests on this topic have referenced this approach. The thick boundary is the union of the outer and inner boundaries as described by Gonzalez and Woods, but the term itself is not present in that text.